File: 07-matrix/01-print-matrix-in-snake-pattern.py
def print_matrix_in_snake_pattern(matrix):
  rows, cols = len(matrix), len(matrix[0])
  for r in range(rows):
    if r % 2 == 0:
      for c in range(cols):
        print(matrix[r][c])
    else:
      for c in range(cols-1, -1, -1):
        print(matrix[r][c])


matrix = [[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12], [13, 14, 15, 16]]
print(print_matrix_in_snake_pattern(matrix))

File: 07-matrix/01-print-matrix-in-snake-pattern/01-print-matrix-in-snake-pattern.rb
def print_matrix_in_snake_pattern(matrix)
  n = matrix.length
  m = matrix[0].length
  res = []
  (0..n-1).each do |i|
    iteration = (0..m-1).to_a
    iteration = (0..m-1).to_a.reverse if i % 2 != 0 
    iteration.each do |j|
      res.append(matrix[i][j])
    end
  end
  res
end


matrix = [[1,2,3,4],[5,6,7,8],[9,10,11,12],[13,14,15,16]]
print print_matrix_in_snake_pattern(matrix)

